src/context/GlobalContext.tsx

```diff
@@ -0,0 +1,62 @@
+import React, {
+  useContext,
+  useReducer,
+  useEffect,
+  createContext,
+  useMemo,
+} from "react";
+import {
+  ContextActions,
+  GlobalState,
+  Types,
+} from "src/context/context.interface";
+import * as ActionCreators from "src/context/ActionCreators";
+import { rootReducer } from "src/context/reducer";
+import { AuthFormData } from "src/common/interface";
+import { auth } from "src/modules/AuthService";
+import { User } from "src/modules/User";
+
+const initialState: GlobalState = {
+  loading: true,
+};
+const GlobalContext = createContext<{
+  state: GlobalState;
+  actions?: ContextActions;
+}>({
+  state: initialState,
+});
+
+export function useAuth() {
+  return useContext(GlobalContext);
+}
+
+export const GlobalProvider: React.FC = ({ children }) => {
+  const [state, dispatch] = useReducer(rootReducer, initialState);
+
+  const actions = useMemo<ContextActions>(
+    () => ({
+      signUp: (payload: AuthFormData) =>
+        ActionCreators.register(dispatch, payload),
+      signIn: (payload: AuthFormData) =>
+        ActionCreators.login(dispatch, payload),
+      signOut: () => ActionCreators.logout(dispatch),
+    }),
+    []
```

Comment:
```suggestion
    [dispatch]
```

Reply:
dispatch won't change over the course of time so not required. Empty dependency array memoized the object and never change it in this case.